pages/docs/tooling/tooling-performance-analysis.md

```diff
@@ -1,70 +1,283 @@
+Measuring the runtime of such operations can be tricky, as the time spend waiting should be ignored.
+Synchronizing all ranks using a barrier solved the issue, but has an impact on performance.
 
-```text
-Run finished at Wed Aug  1 09:41:10 2018
-Global runtime       = 12859ms / 12s
-Number of processors = 4
-# Rank: 0
+To keep this waiting component of the overall measurement to a minimum without affecting performance, we added a configuration option to toggle the synchonization before required events.
+Use the `sync-mode` attribute to enable such synchonization if you need it.
 
-  Event |      Count |  Total[ms] |    Max[ms] |    Min[ms] |    Avg[ms] |       T[%] |
----------------------------------------------------------------------------------------
-_GLOBAL |          1 |      12859 |      12859 |      12859 |      12859 |         99 |
-advance |          1 |         84 |         84 |         84 |         84 |          0 |
+```xml
+<precice-configuration sync-mode="1">
+    ...
+</precice-configuration>
+```
+
+## Configuration
+
+You can configure the profiling using the configuration file with the `<profiling/>` tag located inside `<precice-configuration>`.
+The default settings enable profiling of fundamental events and write the event files the working directory of each solver.
+
+The full configuration consists of:
+
+* `mode` either `off`, `fundamental` (default), or `all`
+* `directory` location to create the `precice-events` folder in and write the event files to
+* the flush frequency  `flush-every`: `0` means only at the end of the simulation, `n>0` means to flush every n-records
+
+### Examples
+
+To turn the profiling off:
+
+```xml
+<precice-configuration>
+  <profiling mode="off" />
+</precice-configuration>
+```
+
+To write all files into a common directory (often `..` works too):
+
+```xml
+<precice-configuration>
+  <profiling director="/path/to/dir" />
+</precice-configuration>
+```
+
+To prevent any write operations of the filesystem until the end of the simulation:
 
-   Name |        Max |  MaxOnRank |        Min |  MinOnRank |    Min/Max |
---------------------------------------------------------------------------
-_GLOBAL |      12859 |          0 |      12859 |          0 |          1 |
-advance |        119 |          2 |         83 |          1 |          0 |
+```xml
+<precice-configuration>
+  <profiling flush-every="0" />
+</precice-configuration>
 ```
 
-`T[%]` prints the relative runtime. Note that this can be more than 100% summed up, since events can be nested, like in the example above.
+To profile blocking operations:
+
+```xml
+<precice-configuration sync-mode="1">
+  <profiling mode="all" />
+</precice-configuration>
+```
+
+## Post-processing workflow
+
+The general workflow looks as follows:
+
+1. Run the simulation
+2. Merge the event files
+3. Analyse each participants that executes a mapping
```

Comment:
Why the focus on mappings?
Mesh repartitioning, acceleration, ... could also get important.

Reply:
Configuring a mapping often implies repartitioning. I added acceleration though.